website/content/docs/v0.9/Guides/converting-control-plane.md

```diff
@@ -0,0 +1,247 @@
+```
+
+Make sure that manifests and static pods are correct across all control plane nodes, as each node reconciles
+control plane state on its own.
+For example, CNI configuration in machine config should be in sync across all the nodes.
+Talos nodes try to create any missing Kubernetes resources from the manifests, but it never
+updates or deletes existing resources.
+
+If something looks wrong, script can be aborted and machine configuration should be updated to fix the problem.
+Once configuration is updated, the script can be restarted.
+
+If static pod definitions and manifests look good, confirm next step to disable `pod-checkpointer`:
+
+```bash
+$ talosctl -n <IP> convert-k8s
+...
+confirm disabling pod-checkpointer to proceed with control plane update [yes/no]: yes
+disabling pod-checkpointer
+deleting daemonset "pod-checkpointer"
+checking for active pod checkpoints
+2021/03/09 23:37:25 retrying error: found 3 active pod checkpoints: [pod-checkpointer-655gc-talos-default-master-3 pod-checkpointer-pw6mv-talos-default-master-1 pod-checkpointer-zdw9z-talos-default-master-2]
+2021/03/09 23:42:25 retrying error: found 1 active pod checkpoints: [pod-checkpointer-pw6mv-talos-default-master-1]
+confirm applying static pod definitions and manifests [yes/no]:
+```
+
+Self-hosted control plane runs `pod-checkpointer` to work around issues with control plane availability.
+It should be disabled before conversion starts to allow self-hosted control plane to be removed.
+It takes around 5 minutes for the `pod-checkpointer` to be fully disabled.
+Script verifies that all checkpoints are removed before proceeding.
+
+Last confirmation before proceeding is at the point when there is no way to keep running self-hosted control plane:
+static pods are released, bootstrap manifests are applied, self-hosted control plane is removed.
+
+```bash
+$ talosctl -n <IP> convert-k8s
+...
+confirm applying static pod definitions and manifests [yes/no]: yes
+removing self-hosted initialized key
+waiting for static pods for "kube-apiserver" to be present in the API server state
+waiting for static pods for "kube-controller-manager" to be present in the API server state
+waiting for static pods for "kube-scheduler" to be present in the API server state
+deleting daemonset "kube-apiserver"
+waiting for static pods for "kube-apiserver" to be present in the API server state
+deleting daemonset "kube-controller-manager"
+waiting for static pods for "kube-controller-manager" to be present in the API server state
+deleting daemonset "kube-scheduler"
+waiting for static pods for "kube-scheduler" to be present in the API server state
+conversion process completed successfully
+```
+
+As soon as control plane static pods are rendered, kubelet starts control plane static pods.
+Pods for `kube-apiserver` are crashing initially, as port :6443 is already occupied by self-hosted `kube-apiserver`.
```

Comment:
"It is expected that the Pods for `kube-apiserver` will crash initially.  Only one `kube-apiserver` can be bound to the host Node's port 6443 at a time.  Eventually, the old `kube-apiserver` will be killed, and the new one will be able to start.  This is all handled automatically."